.ci/azure-pipelines/docs-pipeline.yaml

```diff
@@ -0,0 +1,46 @@
+trigger:
+  paths:
+    include:
+    - doc
+
+pr:
+  paths:
+    include:
+    - doc
+
+resources:
+  pipelines:
+    - pipeline: Build-CI
+      source: Build
+      trigger:
+        stages:
+        - build_gcc
+  containers:
+    - container: fmt # for formatting.yaml
+      image: pointcloudlibrary/fmt
+    - container: doc # for documentation.yaml
+      image: pointcloudlibrary/doc
+    - container: env1804 # for tutorials.yaml
+      image: pointcloudlibrary/env:18.04
+
+stages:
+  - stage: formatting
+    displayName: Formatting
+    # if docs pipeline triggered by build_gcc stage,
+    # the formatting stage has already run, thus it
+    # won't run for a second time here.
+    condition: not(eq(variables['Build.Reason'], 'ResourceTrigger'))
+    jobs:
+      - template: formatting.yaml
+
+  - stage: documentation
+    displayName: Documentation
+    condition: in(dependencies.formatting.result, 'Succeeded', 'SucceededWithIssues', 'Skipped')
+    jobs:
+      - template: documentation.yaml
+
+  - stage: tutorials
+    displayName: Tutorials
+    condition: in(dependencies.documentation.result, 'Succeeded', 'SucceededWithIssues')
```

Comment:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeeded

We can simplify using just `succeeded()` as the condition

Reply:
Thanks for your reviews!

Actually, I've tried `succeeded()`,`always()` expressions in these commits to my test branch `test-new-docs-pipeline`: https://github.com/ueqri/pcl/commit/d4bb60aa4bfb0caf1c07ce82de08c6234598c2f2 https://github.com/ueqri/pcl/commit/691f2425a7f3ae6d6a7ced329e6d4e760285cc5b https://github.com/ueqri/pcl/commit/0ce26835e162b88711390f63e558966f7f944501

But the condition sometime works, and something failed. I guess those `succeeded()`,`always()` are actually `Job status check functions` https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#job-status-functions, but I'm not very sure and find no related official documents about applying job status check functions to stage check. So I turn to https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#dependencies, which was explicitly for stage check.